web/src/engine/osk/src/visualKeyboard.ts

```diff
@@ -326,34 +334,136 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
+
+    // Now to set up event-handling links.
+    // This handler should probably vary based on the keyboard: do we allow roaming touches or not?
+    recognizer.on('inputstart', (source) => {
+      console.log(source);
+      // Make sure we're tracking the source and its currently-selected item (the latter, as we're
+      // highlighting it)
+      const trackingEntry = sourceTrackingMap[source.identifier] = {
+        source: source,
+        roamingHandler: (sample) => {
+          // Maintain highlighting
+          const key = sample.item;
+          const oldKey = sourceTrackingMap[source.identifier].key;
+
+          if(key != oldKey) {
+            this.highlightKey(oldKey, false);
+            this.highlightKey(key, true);
+            sourceTrackingMap[source.identifier].key = key;
+          }
+        },
+        key: source.currentSample.item
+      }
 
-  private get touchInputConfiguration() {
-    let config: InputEventEngineConfig = {
-      targetRoot: this.element,
-      eventRoot: this.element,
-      inputStartHandler: this.touch.bind(this),
-      inputMoveHandler: this.moveOver.bind(this),
-      inputMoveCancelHandler: this.moveCancel.bind(this),
-      inputEndHandler: this.release.bind(this),
-      coordConstrainedWithinInteractiveBounds: this.detectWithinInteractiveBounds.bind(this)
-    };
+      // Yay for closure-capture mechanics:  we can "keep a lock" on this newly-starting
+      // gesture's highlighted key here.
+      this.highlightKey(trackingEntry.key, true);
+
+      const endHighlighting = () => {
+        if(trackingEntry.key) {
+          this.highlightKey(trackingEntry.key, false);
+        }
+      }
+
+      source.path.on('invalidated', endHighlighting);
+      source.path.on('complete', endHighlighting);
+
+      // TODO:  any other 'invalidated' / 'complete' handling needed?
+      // If so, separate handler - it likely needs to be disabled once the first gesture-component
+      // match happens, unlike the highlighting part.
 
-    return new TouchEventEngine(config);
+      source.path.on('step', trackingEntry.roamingHandler);
+
+      source.path.on('step', (sample) => {
+        console.log(`New sample for source ${source.identifier}:`);
+        console.log(sample);
+        // // Do... something based on the potential gesture types that could arise, as appropriate.
+        // // Should be useful for selecting a hint type, etc.
+        // source.potentialModelMatchIds
+      })
+    });
+
+    //
+    recognizer.on('recognizedgesture', (gestureSequence) => {
+      // The highlighting-disablement part of `onRoamingSourceEnd` is 100% safe, so we can leave
+      // that running.
+
+      // Drop any roaming-touch specific behaviors here.
+
+      // This should probably be universal in some manner.
+      gestureSequence.on('complete', () => {
+        console.log("Complete:");
+        console.log(gestureSequence);
+      });
```

Comment:
I take it this is debug temp code? It shouldn't really be in a PR awaiting merge!